### PR TITLE
Create Test Cases for Users API

### PR DESCRIPTION
**Pull request by uyen3893**
Please review it

### CODE REVIEW

File: tests/test-users-api.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+const gender = true
+const email = 'x@email.com'
+const cmnd = '82764619'
+
+
+
+describe("users api", function() {
+    describe('get users', async() => {
+        it("1. should return status code 200 and the number of data response is equal the number of users in database when called correctly", async () => {
+            // axios.get( `${process.env.API_URL}/users`)
```
Comment (phkhuong) [inline tests/test-users-api.js]: Remove redundant code

File: tests/test-users-api.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                const dbUsersCount = result.rows[0].count;
+                assert.strictEqual(apiUsersCount, parseInt(dbUsersCount));
+            })
+            assert.strictEqual(response.status, 200);
+        });
+    })
+
+    describe("get user by ID" , async() => {
+        // var new_user_id;
+        // before(function() {
```
Comment (phkhuong) [inline tests/test-users-api.js]: remove redundant code

File: tests/test-users-api.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        it('4. should return status 500 and response will be null when called with incorrect id\'s value', async() => {
+            var new_user_id;
+            const result = await pool.query('SELECT public."InsertUser"($1, $2, $3, $4, $5, $6, $7) as new_id', 
+            [birthdate, signupdate, name, cmnd, email, gender, expirydate])
+            new_user_id = result.rows[0].new_id.substring(0,36) + '1'
+            console.log("id: " + new_user_id)
+            await axios.get( `${process.env.API_URL}/users/${new_user_id}`).catch(error=>{
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
```
Comment (phkhuong) [inline tests/test-users-api.js]: Description state "response will be null" but the response in code is compared to the not null value.

File: tests/test-users-api.js [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            const response = await axios.post(`${process.env.API_URL}/users`, {
+                name: name,
+                gender: gender,
+                cmnd: cmnd,
+                email: email,
+                birthdate: birthdate.toLocaleDateString(),
+                signupdate: signupdate.toLocaleDateString(),
+                expirydate: expirydate.toLocaleDateString()
+            })
+            console.log('id khuong: ' + response.data.InsertedUserId)
```
Comment (phkhuong) [inline tests/test-users-api.js]: Should remove log

File: tests/test-users-api.js [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+                assert.strictEqual(error.response.status, 500)
+                assert.strictEqual(error.response.data.State, State.ERROR)
+                assert.strictEqual(error.response.data.ErrorMessage, ErrorResponseMessage)
+            })
+            await pool.query('CALL public."DeleteUser" ($1)', [result.rows[0].new_id])
+        })
+    })
+
+    describe('create user', async() => {
+        it('5. should return status 201 and have a different id when called', async() => {
```
Comment (phkhuong) [inline tests/test-users-api.js]: ambiguous description: "have a different id when called"
should change to "successfully create new user"

File: tests/test-users-api.js [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+                const dbUser = result.rows[0];
+                assert.strictEqual(dbUser.name, name);
+                assert.strictEqual(dbUser.cmnd, cmnd);
+                assert.strictEqual(dbUser.gender, gender);
+                assert.strictEqual(dbUser.birthdate.toISOString(), birthdate.toISOString());
+                assert.strictEqual(dbUser.email, email);
+                assert.strictEqual(dbUser.signupdate.toISOString(), signupdate.toISOString());
+                assert.strictEqual(dbUser.expirydate.toISOString(), expirydate.toISOString());
+            })
+            assert.strictEqual(response.status, 201)
```
Comment (phkhuong) [inline tests/test-users-api.js:118]: Should remove inserted user after completed test case

File: tests/test-users-api.js [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+            const updatedCmnd = '325323452'
+            const updatedEmail = 'c@email.com'
+            const updatedGender = true
+            const updatedBirthDate = birthdate
+            updatedBirthDate.setFullYear(birthdate.getFullYear() + 1)
+            const updatedSignUpDate = signupdate
+            updatedSignUpDate.setFullYear(signupdate.getFullYear() + 1)
+            const updatedExpiryDate = expirydate
+            updatedExpiryDate.setFullYear(expirydate.getFullYear() + 1)
+            console.log('expected date: ' + updatedExpiryDate)
```
Comment (phkhuong) [inline tests/test-users-api.js]: should remove log

File: tests/test-users-api.js [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+            const dbUser = updatedResult.rows[0];
+            assert.strictEqual(dbUser.name, updatedName);
+            assert.strictEqual(dbUser.cmnd, updatedCmnd);
+            assert.strictEqual(dbUser.gender, updatedGender);
+            assert.strictEqual(dbUser.birthdate.toISOString(), updatedBirthDate.toISOString());
+            assert.strictEqual(dbUser.email, updatedEmail);
+            assert.strictEqual(dbUser.signupdate.toISOString(), updatedSignUpDate.toISOString());
+            assert.strictEqual(dbUser.expirydate.toISOString(), updatedExpiryDate.toISOString());
+
+            console.log('actual date: ' + dbUser.expirydate)
```
Comment (phkhuong) [inline tests/test-users-api.js]: should remove log

File: tests/test-users-api.js [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+            await pool.query('CALL public."DeleteUser" ($1)', [new_user_id])
+        })
+
+        it('8. should return status 500 and State will be error when called incorrectly', async() => {
+            await axios.put(`${process.env.API_URL}/users/1`, {
+                gender: 'true'
```
Comment (phkhuong) [inline tests/test-users-api.js]: we don't need to put gender here. Do we?

**Review by phkhuong — CHANGES_REQUESTED** [review]

File: tests/test-users-api.js (hunk 1 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 2 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 3 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 4 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 5 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 6 above)
Comment (uyen3893) [reply tests/test-users-api.js:118]: I fixed it

File: tests/test-users-api.js (hunk 7 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 8 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 9 above)
Comment (uyen3893) [reply tests/test-users-api.js]: I fixed it

File: tests/test-users-api.js (hunk 3 above)
Comment (phkhuong) [reply tests/test-users-api.js]: Please remember and review the original purpose of this test case and how you arrange it. Wrong arrange leads to wrong assert, and you've just changed the description following that wrongness.

**Review by phkhuong — APPROVED** [review]

**Pull request merged**